### ISSUE TITLE
Main screen not updating on Loupedeck CT (Hardware Version 02.00)

### ISSUE THREAD

**Issue by thomasnagtegaal40** [fcpxhacks/fcpxhacks#3328]
I recently purchased the Loupedeck CT for Final Cut Pro. I installed CommandPost as indicated on the site. The response of the dials is very slow in Final Cut Pro, especially in the areas of color grading (color wheels, tint, temperature, etc) and scrolling through the timeline. Am I missing something? Any help or tips would be greatly appreciated.

![image](https://github.com/CommandPost/CommandPost/assets/162511601/5a18ebde-2cf6-4351-8133-6a78d16d5b0c)

![image](https://github.com/CommandPost/CommandPost/assets/162511601/2a9e2a78-2baf-4f59-b2d5-69c71222c30b)

![image](https://github.com/CommandPost/CommandPost/assets/162511601/87cf3cba-305b-4be8-8ed7-4665b30cdd40)

**Comment by latenitefilms** [comment]
Are you using CommandPost directly or via the official Loupedeck software as a plugin?

A lot of people have had issues with the official software, so most of our users are just using CommandPost's native/direct support.

**Comment by thomasnagtegaal40** [comment]
I have installed commandpost as described on the website bellow:

https://support.loupedeck.com/setting-up-commandpost-plugin

What do I need to change to use the loupedeck CT via commandpost directly?

Thank you for your quick response!

**Comment by thomasnagtegaal40** [comment]
Never mind, I just found the description. I will give it a try later today.

**Comment by thomasnagtegaal40** [comment]
Tried to use CommandPost's native/direct support today, but it's still lagging quite a bit. With a slow turn on one of the dials it's keeping up, after multiple turns it start lagging. When I use my mouse to control the color wheels for example, and flip it to the sides, it reacts immediately

**Comment by latenitefilms** [comment]
Is the official Loupedeck software definitely not running?

Have you tried a good old fashion restart?

Do you have any other applications running that use the Accessibility API? Check your System Settings/Preferences to see what else uses the Accessibility API, and close them to see if it makes any difference.

**Comment by thomasnagtegaal40** [comment]
Yes restarted multiple times. Next to Commandpost only FCPX had permission to use accessibility API. I turned it off and tried again, but no success. I added some other functions to the dials to do some more testing (move 1 or 10 frames backward or forward via the workflow extension API), but this is even slower. When I turn the dial, first nothing happens, then the Commandpost status window opens up a few times. After a few times it finally says ping received and the playhead start moving, but also really laggy.


https://github.com/CommandPost/CommandPost/assets/162511601/39dabfcb-8fe2-4597-9482-2d3dfc969a17

**Comment by latenitefilms** [comment]
This might seem like a crazy idea, but what cable are you using? Are you using that cable that came with the device?

We have seen some users have all kinds of issues with Loupedeck products when using other USB-C cables.

**Comment by thomasnagtegaal40** [comment]
I use the usb-c cable that comes with the Loupedeck CT and plug it directly into the usb-c port of my MacBook.

**Comment by thomasnagtegaal40** [comment]
I started all over again by reinstalling both the Loupedeck software and Commandpost. Uninstalled Loupedeck as explained on the website, and switched commandpost over from plugin to the native support.

For some reason the input is much more direct now. Might have been a background process that was still running indeed..

**Comment by thomasnagtegaal40** [comment]
However, I now have the problem that my symbols are no longer visible. I have reset all settings and rebooted my MacBook, Commandpost and the loupedeck, but without success.

Also when switching between applications and or switching preview, no symbols are visible.

Any idea what causes this and how to fix it?

![image](https://github.com/CommandPost/CommandPost/assets/162511601/030df277-5353-4b06-ac9e-07bd76b98da9)

![image](https://github.com/CommandPost/CommandPost/assets/162511601/f292870f-b485-4589-9d65-a4e0bdea81dc)

![image](https://github.com/CommandPost/CommandPost/assets/162511601/38b87a31-9922-4af9-b965-824e837c42f6)

![image](https://github.com/CommandPost/CommandPost/assets/162511601/dc706656-4a56-4672-9ceb-ad52dd54edf9)

**Comment by latenitefilms** [comment]
Have the icons ever shown when using CommandPost's Native/Direct support or has it always been blank?

Are you able to share a photo of the Model/Type sticker at the back of the hardware?

What version of the firmware are you running? Is it the latest (via the official Loupedeck software)?

In the past, several times Loupedeck has updated the physical hardware, and changed the way the screens work - so it's entirely possible, if this is a new hardware device, that something has changed - although generally speaking if something like this happens, we normally get a lot of emails from people.

**Comment by thomasnagtegaal40** [comment]
I can't quite remember if the symbols were visible before with the use of direct support, I don't think so. I have attached a picture of the hardware and firmware version.

![image](https://github.com/CommandPost/CommandPost/assets/162511601/aaf4c020-ffd1-4915-a870-ccec151b2952)

![image](https://github.com/CommandPost/CommandPost/assets/162511601/92a29cfa-b011-4ad5-933d-faad71026fb4)

**Comment by thomasnagtegaal40** [comment]
As soon as I install the loupedeck software again and launch it, the symbols from the loupedeck software are displayed right away. But when I switch over to the commandpost software, no symbol is displayed no matter what I try.

**Comment by thomasnagtegaal40** [comment]
Is there anything left to try? I'm getting close to the end of the 30-day trial period. So if I can't get it working, I might be able to return it. Although that would be really unfortunate as I would love to integrate the Loupedeck with CP in my workflow!

**Comment by latenitefilms** [comment]
Try this...

Quit CommandPost, then open this file in a text editor:

```
/Applications/CommandPost.app/Contents/Resources/extensions/hs/loupedeck/init.lua
```

On line 384 it says:

```
self.loupedeckDeviceIsUsingRazerFirmware = false
```

Try changing the `false` to `true`.

Try starting CommandPost again and see if it works.

In the debug console it should say something like:

```
Loupedeck CT or Live connected, so lets check the firmware version...
Loupedeck device is using firmware version: XXX
```

Does it say `0.1.2` like it does in the official Loupedeck app?

I'm currently using Loupedeck CT (Type: LDD-1903, Ver: 01.30), and it's running firmware version v0.2.5.

I'm surprised that you have a newer hardware device, that's running much older firmware?

**Comment by latenitefilms** [comment]
Interestingly, I just noticed on the [Loupedeck website](https://support.loupedeck.com/firmware) that it says:

> Note: Don't be concerned if your firmware displays version 0.1.2. It indicates that you own a more recent Loupedeck CT model equipped with the most up-to-date firmware.

So I think that's definitely the issue - you have new hardware, running new firmware, but we currently don't have any method to detect this, as we were previously relying on the firmware number (i.e. if the firmware is greater than v0.2.5, then we use the newer "Razer" method of writing data to the screen).

Could you take a screenshot of the Loupedeck entry under **Hardware > USB** in **System Information**?

<img width="377" alt="image" src="https://github.com/CommandPost/CommandPost/assets/22286696/1ec12410-e7ce-447b-b73a-e34ba67e981b">

**Comment by latenitefilms** [comment]
FYI - I've reached out to Logitech too, to see if I can get my hands on this newer v02.00 hardware.

**Comment by thomasnagtegaal40** [comment]
Yes it worked, the icons are visible😉 thanks a lot!!!

![image](https://github.com/CommandPost/CommandPost/assets/162511601/d9189ed5-454f-4a58-a4ac-78fb86de85b2)

![image](https://github.com/CommandPost/CommandPost/assets/162511601/32bde6f2-3012-46d9-a967-0e526b964f36)

![image](https://github.com/CommandPost/CommandPost/assets/162511601/7cc1e6ba-d8f7-4a64-bb3c-c45857e33e84)

**Comment by latenitefilms** [comment]
Great news!

To be honest, I'm still not sure of the best way to solve properly at this point. I'm HOPING to get my hands on the physical hardware, to see if I can find some other difference to determine which screen technique to use.

If possible, could you modify that Lua file again and add this code:

```lua
log.df("data: %s", hs.inspect(data))
```

On line 370 after:

```lua
self:requestFirmwareVersion(function(data)
```

When you restart CommandPost, it should then display some data in the Debug Console after "Loupedeck CT or Live connected, so lets check the firmware version...". I'm hoping maybe the `c` or `i` values might be different to the below:

**Loupedeck CT (Type: LDD-1903, Version 01.10) Firmware:**
b = "0.1.79" (Firmware Version)
c = "0.9.0"
i = "1.0.10"

**Loupedeck CT (Type: LDD-1903, Version 01.30) Firmware:**
b = "0.2.5" (Firmware Version)
c = "0.9.0"
i = "1.0.10"

**Comment by thomasnagtegaal40** [comment]
Sorry, I was off for a few days so had no time to look at it again. I will try it this week and sent a reply :)

**Comment by lxne** [comment]
On my Loupedeck Live I had to set `self.loupedeckDeviceIsUsingRazerFirmware = false` to true, too.
Thank you!

Although, it took me quite some time to find this here.
Shouldn't it be mentioned in the Loupedeck help/description?
https://commandpost.io/control-surfaces/loupedeck/#nativedirect-support

Best wishes!

**Comment by latenitefilms** [comment]
This is something I want to fix properly in the app - it's just been a challenge to solve it, as Loupedeck is a bit of a moving target.

**Comment by revphiljackson** [comment]
I am also having this problem. 
Macbook, Séquoia 15.3.2, connected via original lead usb-c, Loupedeck software not installed. 

<img width="997" alt="Image" src="https://github.com/user-attachments/assets/06ccb7cc-9e9a-4d29-9c6c-ee4e299870f7" />

Having read through the above, I could follow as far as looking for the firmware version (I wasn't sure where to find Hardware>USB). I also couldn't see the step that thomasnagtegaal40 said had fixed it. 

I'm also a bit of a novice interms of inserting code. I have done some coding way back but struggling to get my head around the commandpost interface to insert the code provided.  

Any help would be gratefully received.

**Comment by latenitefilms** [comment]
@revphiljackson - Check these instructions:

https://github.com/CommandPost/CommandPost/issues/3328#issuecomment-2017164069

Hope they make sense?

Note to self:

<img width="614" alt="Image" src="https://github.com/user-attachments/assets/45c733a2-7fd7-49ad-80ae-ec3df6bbbc59" />

**Comment by Grassberge** [comment]
Hi, I changed the code on line 374 to the following, given that 0.1.2 is actually a newer version than 0.2.5. 
Is this a good way to go about it?

if firmwareVersion > semver("0.2.5") or firmwareVersion == semver("0.1.2") then -- version 0.1.2 is newer than 0.2.5 - go figure

**Comment by latenitefilms** [comment]
Yeah, quite possibly - I'll try and have a proper look at this over the next few days.

**Comment by ricardoegm-a11y** [comment]
Hello, thank you very much for your support and for addressing this issue. In my case, the problem seems to be only partially resolved. The main dial still does not display any icons. Could you please advise me on how to fix this?

**Comment by thomasnagtegaal40** [comment]
Hello, I recently upgraded to a new MacBook Pro (M4 Pro - Sequoia 15.6.1) and transferred all the applications and files from my previous MacBook Pro. Before starting, I checked the init.lua file to see if the changes were still present, which they were. Unfortunately, after starting Command Post and connecting the Loupedeck, the icons weren't visible. To be safe, I reset the profile and assigned new icons, but they remained grayed out. The debug console showed the following information:

CommandPost v1.5.2 (Build: 8087)

Build Date:                     Dec 18 2024, 17:04:23
macOS Version:                  15.6.1
CommandPost Locale:             en_NL
Current Keyboard Layout:        U.S.
Final Cut Pro Path:             /Applications/Final Cut Pro.app
Final Cut Pro Version:          11.1.1
Final Cut Pro Language:         en
Developer Mode:                 false

2025-09-28 21:36:03: Reloading CommandPost...
2025-09-28 21:36:03: -- Loading /Applications/CommandPost.app/Contents/Resources/extensions/cp/init.lua
2025-09-28 21:36:04: 21:36:04     fcpApp:     Running Final Cut Pro v11.1.1 in a sandbox...
2025-09-28 21:36:04:                          Using sandbox path for preferences: /Users/thomasnagtegaal/Library/Containers/com.apple.FinalCut/Data/Library/Preferences/com.apple.FinalCut
2025-09-28 21:36:04: 
2025-09-28 21:36:04:                  cp:     Loading Plugins...
2025-09-28 21:36:08: 21:36:08 ** Warning:   plugins: Slow Plugin: core.loupedeckctandlive.manager (1.0490560531616)
2025-09-28 21:36:08: 21:36:08 ** Warning:   plugins: Slow Plugin: core.menu.menuaction (1.3030610084534)
2025-09-28 21:36:09: 21:36:09         cp:     Plugins Loaded.
2025-09-28 21:36:09:                          Startup Time: 4.909304 seconds
2025-09-28 21:36:09:                          Garbage Collection Mode: generational
2025-09-28 21:36:09: -- Done.
2025-09-28 21:36:09:           loupedeck:     Loupedeck CT or Live connected, so lets check the firmware version...
2025-09-28 21:36:09: -- Loading extension: hs.inspect
2025-09-28 21:36:09:                          data: {
  b = "0.1.2",
  c = "0.9.0",
  data = "\t",
  i = "0.0.0",
  id = 3079
}
2025-09-28 21:36:09:                          Loupedeck device is using firmware version: v0.1.2

**Comment by lxne** [comment]
Your mac and macOS are probably not causing the issue. I have a M1 Max, everything worked on Sequia and is now working on Tahoe, too. Loupedeck Live (newest firmware).

Have you tried this?

> Quit CommandPost, then open this file in a text editor:
> 
> /Applications/CommandPost.app/Contents/Resources/extensions/hs/loupedeck/init.lua
> 
> On line 384 it says:
> 
> self.loupedeckDeviceIsUsingRazerFirmware = false
> 
> Try changing the false to true.
> 
> Try starting CommandPost again and see if it works.

**Comment by thomasnagtegaal40** [comment]
Yes, I copied everything over from my old Mac and line 384 in the initial.lua file was set to 'true'. 
I did a clean install today of commandpost and changed the init.lua file, and the icons seem to be back! :)

**Comment by TheRealSimon42** [comment]
Just FYI: Setting Line 384 in the initial.lua to `true` did also work for me. Icons are displayed again on my Loupedeck CT 👍

**Comment by Guillaume-Petitcollin** [comment]
> Hello, thank you very much for your support and for addressing this issue. In my case, the problem seems to be only partially resolved. The main dial still does not display any icons. Could you please advise me on how to fix this?

+1 
after setting line 384, main screen is working properly, still not the dial one.
